DICOM/src/Microsoft.Health.DeID.SharedLib/Settings/EncryptSetting.cs

```diff
@@ -0,0 +1,52 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Health.Dicom.DeID.SharedLib.Exceptions;
+
+namespace Microsoft.Health.DeID.SharedLib.Settings
+{
+    public class EncryptSetting
+    {
+        public string EncryptKey { get; set; }
+
+        public string PublicKey { get; set; }
+
+        public string PrivateKey { get; set; }
+
+        public void Validate()
+        {
+            using Aes aes = Aes.Create();
+            var encryptKeySize = Encoding.UTF8.GetByteCount(EncryptKey) * 8;
```

Comment:
Two problems here:
(1) Using a plain string as the encryption key in configuration.json limits the content of key.
Actually in most cases a key is a byte array instead of a printable string.
If you want to specify a key in a text file like json/yaml, you should use proper encoding method like base64.

(2) Encoding.UTF8.GetByteCount doesn't make sense here.
If we allow normal binary key, then we should use proper method to decode the string and get the bytes.
If we intentionally limit the key to printable (letter + digit ?) string, then we can simply calculate the key size by EncryptKey.Length * 8

Overall I think the assumption (use printable characters as key) is unreasonable.

Reply:
add some of my comments. Putting a readable key in your settings is OK to me, but that doesn't mean to use UKF8 to encoding it. I guess your original idea is that when the key is "123", you want to encrypt the data with the key (in bytes array in memory) as "0x1, 0x2, 0x3", right? Anyway, a common way to do that is you need to create a function which can transfer a string to the byte array. For instance, if the real key is:

byte[] key =
            {
                0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
                0x09, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16
            };

your key string in seting is: "01020304050607080910111213141516"

Reply:
Yes, we need an encode/decode method and it should be described in user doc, because users need to understand this to correctly put it into the configuration file.

From current implementation, the logic is not like what @hippogr just mentioned. It uses the ASCII code of each character in the key string as the real key. It's not a commonly used method.

I personally am fine with either base64 encoding or straightforward encoding like @hippogr's suggestion.
The format used by similar tool or key generation tool might be a good reference.

Reply:
To keep consistent with FHIR-Anonymization-Tool released before, we still use UTF-8 decoding for now.
We can change this encoding method for two tools later.